rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -291,10 +292,22 @@ public ResponseEntity<EntityModel<Component>> getComponent(
             @PathVariable("id") String id
     ) throws TException {
         User user = restControllerHelper.getSw360UserFromAuthentication();
-        Component sw360Component = componentService.getComponentForUserById(id, user);
-        HalResource<Component> userHalResource = createHalComponent(sw360Component, user);
-        restControllerHelper.addEmbeddedDataToComponent(userHalResource, sw360Component);
-        return new ResponseEntity<>(userHalResource, HttpStatus.OK);
+        try {
+            Component sw360Component = componentService.getComponentForUserById(id, user);
+            HalResource<Component> userHalResource = createHalComponent(sw360Component, user);
+            restControllerHelper.addEmbeddedDataToComponent(userHalResource, sw360Component);
+            return new ResponseEntity<>(userHalResource, HttpStatus.OK);
+        } catch (SW360Exception e) {
+            Map<String, Object> errorBody = new HashMap<>();
+            errorBody.put("message", e.getWhy());
+            errorBody.put("status", e.isSetErrorCode() ? e.getErrorCode() : 500);
+
+            if (e.isSetErrorCode() && e.getErrorCode() == 404) {
+                return new ResponseEntity(errorBody, HttpStatus.NOT_FOUND);
+            } else {
+                return new ResponseEntity(errorBody, HttpStatus.INTERNAL_SERVER_ERROR);
+            }
```

Comment:
????

Why are we not throwing `NoResourceFoundException` or `SW360Exception` here to get proper response object?